SPICE colormaps depending on temperature

adolliou opened the pull request:
implemented an util/cmap.py code with a get_cmap function that returns a colormap according to the line temperature. The cmap is designed with lab and a combines a linear scale with a gray scale and strong weights on the boundaries.

Added also preliminary tests on the cmap instance.

adolliou committed:
implemented an util/cmap.py code with a get_cmap function that returns a colormap according to the line temperature. The cmap is designed with lab and a combines a linear scale with a gray scale and strong weights on the boundaries

co-authored-by: Eric Buchlin <eric.buchlin@cnrs.fr>

ebuchlin committed:
Add skimage requirement

ebuchlin committed:
Code reformatting, update pre-commit packages

ebuchlin committed:
Update

ebuchlin committed:
Add "spice" in name

ebuchlin committed:
Add tests

ebuchlin committed:
Make get_cmap available as sospice.get_cmap()

ebuchlin commented:
I've added some more tests, ensured that pre-commit works, and made the function available as `sospice.get_cmap()`.

Some ideas:

* The behavior when `logt` is out of [4, 6] could be something else than throwing an exception. Why not just clipping `logt_fraction` to [0, 1] in `_get_main_color()`? If users want to get a color table for logt = 2 or 7, they would get the same color table as for logt = 4 or 6, but is that a problem?
* The color table parameters `logt_range, saturation, hue_factor` and their default values could be moved from  `_get_main_color()` to `get_cmap()`, so that they can be changed by users.

In the end we should ensure to merge the changes to develop and not to main (main is only when we do a release).

adolliou committed:
cippled value of logt_range between 0 and 1 to ensure that colomaps with logt not between (4, 6) can also be created,  and added the parameters of _get_main_color in get_cmap as inputs

adolliou commented:
Hi

> I've added some more tests, ensured that pre-commit works, and made the function available as `sospice.get_cmap()`.
> 
> Some ideas:
> 
>     * The behavior when `logt` is out of [4, 6] could be something else than throwing an exception. Why not just clipping `logt_fraction` to [0, 1] in `_get_main_color()`? If users want to get a color table for logt = 2 or 7, they would get the same color table as for logt = 4 or 6, but is that a problem?
> 
>     * The color table parameters `logt_range, saturation, hue_factor` and their default values could be moved from  `_get_main_color()` to `get_cmap()`, so that they can be changed by users.
> 
> 
> In the end we should ensure to merge the changes to develop and not to main (main is only when we do a release).

Hi, 

I implemented these changes. The next step would be add to the test the clipping of the logt value.

ebuchlin committed:
Formatting, doc updates, no defaults for private function parameters

ebuchlin committed:
Merge branch 'develop' into main
